src/snowflake/snowpark/_internal/ast_utils.py

```diff
@@ -570,9 +583,11 @@ def build_expr_from_snowpark_column_or_table_fn(
     if isinstance(value, snowflake.snowpark.Column):
         build_expr_from_snowpark_column(expr_builder, value)
     elif isinstance(value, snowflake.snowpark.table_function.TableFunctionCall):
-        raise NotImplementedError(
-            "SNOW-1509198: No support for TableFunctionCall AST generation"
-        )
+        assert value._ast is not None, "TableFunctionCall must have ast assigned."
+        expr_builder.CopyFrom(value._ast)
+        # raise NotImplementedError(
```

Comment:
Is this fully implemented now, or should this branch still raise at the end?

Reply:
I think so, there are not very exhaustive tests for this though. Let me leave a TODO with tests.

Reply:
added comment.